### PR TITLE
modifications to disvis webserver tutorial

### PR DESCRIPTION
**Pull request by schaarj**
my modifications to the tutorial, some further comments/suggestions:

* `/path/to/` might be easily copied without modification and not applicable to windows (maybe delete  the commands and just mention that a command to open the files exists)
* randomise position of partners so that they are not in contact when loaded
* first color residues of crosslink to highlight patches
* visualisation of distances might be more applicable in a predicted complex
* maybe add instructions to play with radio buttons and expand advanced options group
* visualising AIS with 7 restraints shows nothing in chimera in the example (maybe also add chimera screenshot with Volume Viewer window)
* Add reference to "conformation in the literature”
* Explain Cutoff for defining a residue as important/active (the 10 and 12 answer doesn’t seem self explanatory to me)

### CODE REVIEW

**Comment by amjjbonvin** [comment]
> randomise position of partners so that they are not in contact when loaded

But then they can not longer guess which restraints are false positives… We do ask for it in the tutorial.
Also at the later stage to compare the ones identified by disvis with their initial guess.

Not ideal I agree, but kind of ok for the purpose of the tutorial

**Comment by amjjbonvin** [comment]
> visualisation of distances might be more applicable in a predicted complex

Still nice to show how to do it - since we do not predict the complex in this tutorial…

**Comment by mtrellet** [comment]
Thank you for the feedback and the time you dedicated to it! Here are few answers:

> /path/to/ might be easily copied without modification and not applicable to windows (maybe delete the commands and just mention that a command to open the files exists)

Yes, followed example for the powerfit web server, we can change that even if I think that users will figure out that's they need to change the path. Why would it not work on windows? There's a command line tool also there and I just tested, works perfectly.

> randomise position of partners so that they are not in contact when loaded
...
> visualisation of distances might be more applicable in a predicted complex

Seems contradictory... Or we add a fitting step. We do it at the end of the tutorial. Anyway, PRE5 and PUP2 are placed as they are found in the homolog model 1ryp (chain E+F). It's also as we provide them in the example, I suppose that it comes from the homolog model evoked in Gydo's paper. I cannot be 100% sure since the supp. mat. of the paper, where the model is supposed to be, is not accessible.

> first color residues of crosslink to highlight patches

Rather easy to do, I'll add this step.

> maybe add instructions to play with radio buttons and expand advanced options group

I can add a sentence like that when I'm talking about the sampling parameters.

> visualising AIS with 7 restraints shows nothing in chimera in the example (maybe also add chimera screenshot with Volume Viewer window)

I can clearly see it in both the web page and the output files (Tutorial2.tgz). You were talking about the accessible_interaction_space.mrc, right?

> Add reference to "conformation in the literature”

I'd rather change the whole question, this made sense when I was evoking the crystal structure in the first step but it has moved to the last steps. Good catch ;)

> Explain Cutoff for defining a residue as important/active (the 10 and 12 answer doesn’t seem self explanatory to me)

I'll add the cutoff indeed, rather important even if empirical.

To sum up, I'll add the changes I mentioned above asap. In the meantime, discussions are open..!

**Comment by schaarj** [comment]
>> /path/to/ might be easily copied without modification and not applicable to windows (maybe delete  the commands and just mention that a command to open the files exists)  

> Yes, followed example for the powerfit web server, we can change that even if I think that users will  figure out that's they need to change the path. Why would it not work on windows? There's a command line tool also there and I just tested, works perfectly.

Just wasn't sure how windows treats paths and what the correct format in chimera would be than `c:\\...` ? but if you can test that

>> randomise position of partners so that they are not in contact when loaded
>> ...
>> visualisation of distances might be more applicable in a predicted complex

> Seems contradictory... Or we add a fitting step. We do it at the end of the tutorial. Anyway, PRE5 and PUP2 are placed as they are found in the homolog model 1ryp (chain E+F). It's also as we provide them in the example, I suppose that it comes from the homolog model evoked in Gydo's paper. I cannot be 100% sure since the supp. mat. of the paper, where the model is supposed to be, is not accessible.

I'm thinking about the impact of DisVis here. As the general use case should be starting with two partners not already in a complexed position. I would thus advocate randomising the starting orientation and just visualizing  the position of the croslinked residues (Do they cluster, are they surface accessible) first  and display the distances after we did the fitting to the homologous  structure.

>> maybe add instructions to play with radio buttons and expand advanced options group

> I can add a sentence like that when I'm talking about the sampling parameters.

already added something similar after writing this comment - just check it won't be superfluous

>> visualising AIS with 7 restraints shows nothing in chimera in the example (maybe also add chimera screenshot with Volume Viewer window)

> I can clearly see it in both the web page and the output files (Tutorial2.tgz). You were talking about the accessible_interaction_space.mrc, right?

Yes I was using a quick scanning run I submitted - Did your example come from complete scanning?

>> Add reference to "conformation in the literature”

> I'd rather change the whole question, this made sense when I was evoking the crystal structure in the first step but it has moved to the last steps. Good catch ;)

maybe remove this altogether especially if we go for the above mentioned changes

>> Explain Cutoff for defining a residue as important/active (the 10 and 12 answer doesn’t seem self explanatory to me)

> I'll add the cutoff indeed, rather important even if empirical.

A cutoff of 1 might work for this example but with less stringent restraints it probably won't (See example 1). I would refrain from mentioning something  which sounds like best practice (as in the last commit to your branch) and just suggest to look at the top N for now until we have some benchmark on this.

> To sum up, I'll add the changes I mentioned above asap. In the meantime, discussions are open..!

**Comment by mtrellet** [comment]
> I'm thinking about the impact of DisVis here. As the general use case should be starting with two partners not already in a complexed position. I would thus advocate randomising the starting orientation and just visualizing the position of the croslinked residues (Do they cluster, are they surface accessible) first and display the distances after we did the fitting to the homologous structure.

Why not, makes sense to me, @amjjbonvin , what's your take on this one? We can show the cross-links at the end, once we superimposed the models on S. cerevisiae homolog. We then check if both active residues and cross-links are coherent. And we keep only a generic overview of the regions where the cross-links come from.

> Yes I was using a quick scanning run I submitted - Did your example come from complete scanning?

As said in the tutorial, yes ;)

> maybe remove this altogether especially if we go for the above mentioned changes

il will be removed if we process as stated above

> A cutoff of 1 might work for this example but with less stringent restraints it probably won't (See example 1). I would refrain from mentioning something which sounds like best practice (as in the last commit to your branch) and just suggest to look at the top N for now until we have some benchmark on this.

But top N is completely case-dependant and will vary with the size of the list provided, the size of the interface, of the partners, etc. 
At least 1 has a minimum of relevance since you're looking for residues in contact... If you find one residue making at least one contact in average, you can be quite confident that he's at the interface (providing you have enough complex of course, less trustful for only a few complex, but we have ~20K complexes in our example.)
